packages/flutter_tools/lib/src/commands/widget_preview.dart

```diff
@@ -125,51 +158,235 @@ final class WidgetPreviewStartCommand extends WidgetPreviewSubCommandBase with C
+    widgetPreviewScaffold.createSync();
+
+    if (generateScaffoldProject) {
       logger.printStatus('Creating widget preview scaffolding at: ${widgetPreviewScaffold.path}');
       await generateApp(
-        <String>['widget_preview_scaffold'],
+        <String>['app', kWidgetPreviewScaffoldName],
         widgetPreviewScaffold,
         createTemplateContext(
           organization: 'flutter',
-          projectName: 'widget_preview_scaffold',
+          projectName: kWidgetPreviewScaffoldName,
           titleCaseProjectName: 'Widget Preview Scaffold',
           flutterRoot: Cache.flutterRoot!,
           dartSdkVersionBounds: '^${cache.dartSdkBuild}',
-          linux: const LocalPlatform().isLinux,
-          macos: const LocalPlatform().isMacOS,
-          windows: const LocalPlatform().isWindows,
+          linux: platform.isLinux,
+          macos: platform.isMacOS,
+          windows: platform.isWindows,
         ),
         overwrite: true,
         generateMetadata: false,
       );
-      await _populatePreviewPubspec(rootProject: rootProject);
+
+      // WARNING: this access of widgetPreviewScaffoldProject needs to happen after we generate the
+      // scaffold project as invoking the getter triggers lazy initialization of the preview scaffold's
+      // FlutterManifest before the scaffold project's pubspec has been generated.
+      // TODO(bkonyi): add logic to rebuild after SDK updates
+      await initialBuild(widgetPreviewScaffoldProject: rootProject.widgetPreviewScaffoldProject);
     }
 
-    // WARNING: this needs to happen after we generate the scaffold project as invoking the
-    // widgetPreviewScaffoldProject getter triggers lazy initialization of the preview scaffold's
-    // FlutterManifest before the scaffold project's pubspec has been generated.
     _previewCodeGenerator = PreviewCodeGenerator(
       widgetPreviewScaffoldProject: rootProject.widgetPreviewScaffoldProject,
       fs: fs,
     );
 
+    // TODO(matanlurey): Remove this comment once flutter_gen is removed.
+    //
+    // Tracking removal: https://github.com/flutter/flutter/issues/102983.
+    //
+    // Populate the pubspec after the initial build to avoid blowing away the package_config.json
+    // which may have manual changes for flutter_gen support.
+    await _populatePreviewPubspec(rootProject: rootProject);
+
     final PreviewMapping initialPreviews = await _previewDetector.initialize(rootProject.directory);
     _previewCodeGenerator.populatePreviewsInGeneratedPreviewScaffold(initialPreviews);
 
+    if (launchPreviewer) {
+      shutdownHooks.addShutdownHook(() async {
+        await _widgetPreviewApp?.stop();
+      });
+      _widgetPreviewApp = await runPreviewEnvironment(
+        widgetPreviewScaffoldProject: rootProject.widgetPreviewScaffoldProject,
+      );
+      final int result = await _widgetPreviewApp!.runner.waitForAppToFinish();
+      if (result != 0) {
+        throwToolExit(null, exitCode: result);
```

Comment:
We shouldn't ever throw a tool exit without a message. The user won't see any error message (unless we print one elsewhere). It's also confusing to troubleshoot failing tests that run into this scenario, IIRC.

https://github.com/flutter/flutter/blob/5491c8c146441d3126aff91beaa3fb5df6d710d0/packages/flutter_tools/lib/src/base/common.dart#L24-L26

I lost the original context for this, unfortunately.

Reply:
Good catch, I think I meant to come back and provide a more meaningful message here.